lib/auth/bot_test.go

```diff
@@ -307,3 +307,64 @@ func TestRegisterBotCertificateGenerationStolen(t *testing.T) {
 	require.NoError(t, err)
 	require.NotEmpty(t, locks)
 }
+
+// TestRegisterBotCertificateGenerationStolen simulates a stolen renewable
+// certificate where a generation check is expected to fail.
+func TestRegisterBotCertificateGenerationStolenNew(t *testing.T) {
```

Comment:
We should remove this before merge, right?

Reply:
Yes, there are two commits marked for removal in their descriptions.